src/agent/onefuzz/src/az_copy.rs

```diff
@@ -130,13 +131,50 @@ fn should_always_retry(err: &anyhow::Error) -> bool {
     false
 }
 
+fn file_change_detected(
+    file: &OsStr,
+    pre_copy: Result<Metadata, std::io::Error>,
+    post_copy: Result<Metadata, std::io::Error>,
+) -> bool {
+    match (pre_copy, post_copy) {
+        (Ok(pre_meta), Ok(post_meta)) => {
+            let pre_len = pre_meta.len();
+            let post_len = post_meta.len();
+            if pre_meta.len() != post_meta.len() {
+                info!(
+                    "File length changed from {} to {} for file {:?}",
+                    pre_len, post_len, file
+                );
+                true
+            } else {
+                match (pre_meta.modified(), post_meta.modified()) {
+                    (Ok(pre_modified), Ok(post_modified)) => {
+                        if pre_modified != post_modified {
+                            info!(
+                                "File modified timestamp changed from {:?} to {:?} for file {:?}",
+                                pre_modified, post_modified, file
+                            );
+                            true
+                        } else {
+                            false
+                        }
+                    }
+                    (_, _) => false,
+                }
+            }
+        }
+        (_, _) => false,
```

Comment:
1. In Rust, you can just use `_` to pattern match on the entire tuple, which makes more sense since we're just discarding the match completely, and we aren't disambiguating anything by matching on tuple positions.
2. I think it'd be more idiomatic Rust to do an early return in the one case where we're checking values, and just default to `false` as the non-early return path. You could do an `if let` to just match on that one situation, but as noted above, I think this function should probably just take two `Metadata` values, or maybe `&Metadata`.